src/main/java/uk/gov/hmcts/ccd/security/idam/IdamRepository.java

```diff
@@ -66,19 +65,13 @@ public String getDataStoreSystemUserAccessToken() {
             applicationParams.getDataStoreSystemUserPassword());
     }
 
-    private HttpStatus getHttpStatus(FeignException exception) {
-        HttpStatus httpStatus = HttpStatus.valueOf(exception.status());
+    private RuntimeException parseFeignException(FeignException exception) {
+        boolean isClientError = exception.status() >= 400 && exception.status() <= 499;
 
-        if (exception instanceof FeignException.FeignClientException) {
-            if (httpStatus != HttpStatus.UNAUTHORIZED) {
-                httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
-            }
-        } else if (exception instanceof FeignException.FeignServerException) {
-            if (httpStatus == HttpStatus.INTERNAL_SERVER_ERROR) {
-                httpStatus = HttpStatus.BAD_GATEWAY;
-            }
+        if (isClientError) {
+            return new InvalidTokenException(exception.getMessage(), exception);
```

Comment:
Can you throw the exceptions from here itself?

Reply:
Throwing exception directly from here would change the caller method's return structure as well. I removed **parseFeignException** method and moved its flow to the catch block.